### DISCUSSION TITLE
Chargement du graphique par filtre XML

### DISCUSSION THREAD

**Discussion by NicolasGK** [mviewer/mviewer#947] [Q&A]
Bonjour, 

J'ai actuellement une application qui affiche des plans d'eau pour les années de 2020 à 2023, j'ai un fichier template associé à ma couche qui affiche un graphique d'évolution pour chaque année. J'ai un filtre sur mon XML qui affiche un menu déroulant pour chaque année, ce qui me permet dans mon template .mst de n'afficher que les données pour l'année sélectionnée. 

Maintenant j'aimerais que mon graphique se mette à jour dès que je change d'année dans le menu déroulant (géré par le xml) pour que l'utilisateur n'ait pas à recliquer sur le plan d'eau pour afficher les bonnes données. 

Pour le moment j'arrive à recharger le graphique, qui prend bien en tire l' {{annee}} sélectionnée dans le menu déroulant mais affiche toujours les données de la première année sur laquelle j'ai cliqué : 

```
// Ajouter le gestionnaire d'événements pour le changement de sélection d'année
        document.getElementById('attributes-selector').addEventListener('change', function () {
            var selectedYear = this.value;

            if (parseInt(selectedYear) !== parseInt(currentYear)) {
                currentYear = selectedYear;  // Mise à jour l'année dans la variable currentYear

                // Rechargement du graphique
                drawChart(currentYear);
            }
        });

```        
Auriez-vous des pistes pour rendre cela possible ? 

Merci par avance :)

**Comment by lecault** [comment]
Bonjour,
Nous utilisons un comportement similaire au niveau de la région Bretagne mais nous n'avons pas chercher à rendre le changement interactif faute de compétence en développement.
En effet, ça serait intéressant que cela fonctionne comme vous le désirez.
